File: afc_saaspro_tax/afc_rest_apis/javascript/scripts/model/Exclusion.js
/**
 * Specifies a tax exclusion by state.
 */

function Exclusion() {

    this.CountryISO = "";
    this.State = "";
    this.ExclusionOn = false;

    /**
     * Sets the three-character country ISO code. Required if State not specified.
     * @param countryISO String
     * @return Reference to this object for method chaining. 
     */
    this.countryISO = function (countryISO) {
        this.CountryISO = countryISO;
        return this;
    }

    /**
     * Three-character country ISO code. Required if State not specified.
     * @return String
     */
    this.getCountryISO = function () {
        return this.CountryISO;
    }

    /* Reference to this object for method chaining.*
     * Sets the three-character country ISO code. Required if State not specified.
     * @param countryISO String 
     */
    this.setCountryISO = function (countryISO) {
        this.CountryISO = countryISO;
    }

    /**
     * Sets the two-character state abbreviation. Required if CountryISO not specified.
     * @param state String
     * @return Reference to this object for method chaining. 
     */
    this.state = function (state) {
        this.State = state;
        return this;
    }

    /**
     * Two-character state abbreviation. Required if CountryISO not specified.
     * @return String
     */
    this.getState = function () {
        return this.State;
    }

    /* Reference to this object for method chaining.*
     * Sets the two-character state abbreviation. Required if CountryISO not specified.
     * @param state String 
     */
    this.setState = function (state) {
        this.State = state;
    }

    /**
     * Sets a value indicating if the exclusion applies in the state.
     * @param exclusionOn Boolean
     * @return Reference to this object for method chaining. 
     */
    this.exclusionOn = function (exclusionOn) {
        this.ExclusionOn = exclusionOn;
        return this;
    }

    /**
     * True if entity exclusion applies in the state, otherwise false.
     * @return Boolean
     */
    this.getExclusionOn = function () {
        return this.ExclusionOn;
    }

    /* Reference to this object for method chaining.*
     * Sets a value indicating if the exclusion applies in the state.
     * @param exclusionOn Boolean 
     */
    this.setExclusionOn = function (exclusionOn) {
        this.ExclusionOn = exclusionOn;
    }
}

File: afc_saaspro_tax/afc_rest_apis/java/src/avalara/comms/rest/sample/model/Exclusion.java
package avalara.comms.rest.sample.model;

import com.google.gson.annotations.SerializedName;

/**
 * Specifies a tax exclusion by state.
 */
public class Exclusion {

    @SerializedName("CountryISO")
    private String countryISO = null;

    @SerializedName("State")
    private String state = null;

    @SerializedName("ExclusionOn")
    private Boolean exclusionOn = null;

    /**
     * Sets the three-character country ISO code. Required if State not specified.
     * @param countryISO
     * @return 
     */
    public Exclusion countryISO(String countryISO) {
        this.countryISO = countryISO;
        return this;
    }

    /**
     * Three-character country ISO code. Required if State not specified.
     * @return countryISO
     */
    public String getCountryISO() {
        return countryISO;
    }

    /**
     * Sets the three-character country ISO code. Required if State not specified.
     * @param countryISO 
     */
    public void setCountryISO(String countryISO) {
        this.countryISO = countryISO;
    }

    /**
     * Sets the two-character state abbreviation. Required if CountryISO not specified.
     * @param state
     * @return 
     */
    public Exclusion state(String state) {
        this.state = state;
        return this;
    }

    /**
     * Two-character state abbreviation. Required if CountryISO not specified.
     * @return state
     */
    public String getState() {
        return state;
    }

    /**
     * Sets the two-character state abbreviation. Required if CountryISO not specified.
     * @param state 
     */
    public void setState(String state) {
        this.state = state;
    }

    /**
     * Sets a value indicating if the exclusion applies in the state.
     * @param exclusionOn
     * @return 
     */
    public Exclusion exclusionOn(Boolean exclusionOn) {
        this.exclusionOn = exclusionOn;
        return this;
    }

    /**
     * True if entity exclusion applies in the state, otherwise false.
     * @return exclusionOn
     */
    public Boolean getExclusionOn() {
        return exclusionOn;
    }

    /**
     * Sets a value indicating if the exclusion applies in the state.
     * @param exclusionOn 
     */
    public void setExclusionOn(Boolean exclusionOn) {
        this.exclusionOn = exclusionOn;
    }
}
